am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java

```diff
@@ -350,4 +362,236 @@ private DefaultPermissionGrant createDefaultPermissionGrant(String roleName,
+                .build())
+            .access(AccessEnvelope.builder()
+                .permissions(ImmutableMap.of(
+                    JsonPointer.valueOf(""), ImmutableSet.of(READ),
+                    JsonPointer.valueOf(NAME_ATTRIBUTE), ImmutableSet.of(READ),
+                    JsonPointer.valueOf(CITY_ATTRIBUTE), ImmutableSet.of(READ)))
+                .accessType(ROLE_BASED)
+                .build())
+            .relationships(ImmutableSet.of())
+            .build());
+    }
+
+    //This method checks if the User's security role has less privileges as of the requested attributes.
+    //In this case, the filter method should return no attribute.
+    @Test
+    void whenRoleSecurityClassificationIsLessThanResourceSecurityClassificationShouldReturnNull() {
+        String idamRoleWithRoleBasedPrivateAccess = UUID.randomUUID().toString();
+        importerService.addRole(idamRoleWithRoleBasedPrivateAccess, IDAM, PUBLIC, ROLE_BASED);
+
+        Map<JsonPointer, SecurityClassification> attributePermissions = new ConcurrentHashMap<>();
+        attributePermissions.put(JsonPointer.valueOf(""), PRIVATE);
+        attributePermissions.put(JsonPointer.valueOf(NAME_ATTRIBUTE), PRIVATE);
+        attributePermissions.put(JsonPointer.valueOf(CITY_ATTRIBUTE), PRIVATE);
+
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(idamRoleWithRoleBasedPrivateAccess,
+            resourceDefinition, "", ImmutableSet.of(READ)));
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(idamRoleWithRoleBasedPrivateAccess,
+            resourceDefinition, NAME_ATTRIBUTE, ImmutableSet.of(READ)));
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(idamRoleWithRoleBasedPrivateAccess,
+            resourceDefinition, CITY_ATTRIBUTE, ImmutableSet.of(READ)));
+
+        FilteredResourceEnvelope result = service.filterResource(accessorId, ImmutableSet.of(idamRoleWithRoleBasedPrivateAccess),
+            createResource(resourceId, resourceDefinition), attributePermissions);
+
+        assertThat(result).isEqualTo(FilteredResourceEnvelope.builder()
+            .resource(Resource.builder()
+                .id(resourceId)
+                .definition(resourceDefinition)
+                .data(null)
+                .build())
+            .access(AccessEnvelope.builder()
+                .permissions(ImmutableMap.of())
+                .accessType(ROLE_BASED)
+                .build())
+            .relationships(ImmutableSet.of())
+            .build());
+    }
+
+    //This method checks if few of the security classification attributes have higher privilege than the user security classification.
+    //In this case, the filter method should return the attributes for which user is authorized.
+    @Test
+    void whenRoleSecurityClassificationIsLessThanAttributeSecurityClassificationShouldRemoveAttribute() {
+        //Permissions are looking fine, data needs to be updated.
```

Comment:
remove comment? same for others below

Reply:
Removed the comments from all of the test cases.